File: src/main/java/eladkay/quaritum/common/networking/NetworkHelper.java
package eladkay.quaritum.common.networking;

import eladkay.quaritum.api.lib.LibMisc;
import net.minecraft.util.math.BlockPos;
import net.minecraftforge.fml.common.network.NetworkRegistry;
import net.minecraftforge.fml.common.network.simpleimpl.IMessage;
import net.minecraftforge.fml.common.network.simpleimpl.SimpleNetworkWrapper;
import net.minecraftforge.fml.relauncher.Side;

public class NetworkHelper {
    public static SimpleNetworkWrapper instance;

    public static void init() {
        instance = new SimpleNetworkWrapper(LibMisc.MOD_ID);
        try {
            instance.registerMessage(FancyParticlePacket.class, FancyParticlePacket.class, 0, Side.CLIENT);
            instance.registerMessage(LightningEffectPacket.class, LightningEffectPacket.class, 1, Side.CLIENT);
            instance.registerMessage(FancyLineParticlePacket.class, FancyLineParticlePacket.class, 2, Side.CLIENT);
        } catch (NoClassDefFoundError server) {
            server.printStackTrace();
        }
        //todo this is bad
        //Caused by: java.lang.NoClassDefFoundError: net/minecraft/client/entity/EntityPlayerSP
    }

    //lol
    public static void tellEveryone(IMessage message) {
        instance.sendToAll(message);
    }

    public static void tellEveryoneAround(IMessage message, int dim, int x, int y, int z, int range) {
        instance.sendToAllAround(message, new NetworkRegistry.TargetPoint(dim, x, y, z, range));
    }

    public static void tellEveryoneAround(IMessage message, int dim, BlockPos pos, int range) {
        instance.sendToAllAround(message, new NetworkRegistry.TargetPoint(dim, pos.getX(), pos.getY(), pos.getZ(), range));
    }

    public static void tellEveryoneAround(IMessage message, NetworkRegistry.TargetPoint point) {
        instance.sendToAllAround(message, point);
    }
}

File: src/main/java/eladkay/quaritum/common/networking/NetworkHelper.kt
package eladkay.quaritum.common.networking

import eladkay.quaritum.api.lib.LibMisc
import net.minecraft.util.math.BlockPos
import net.minecraftforge.fml.common.network.NetworkRegistry
import net.minecraftforge.fml.common.network.simpleimpl.IMessage
import net.minecraftforge.fml.common.network.simpleimpl.SimpleNetworkWrapper
import net.minecraftforge.fml.relauncher.Side

object NetworkHelper {
    var instance: SimpleNetworkWrapper

    init {
        instance = SimpleNetworkWrapper(LibMisc.MOD_ID)
        try {
            instance.registerMessage<FancyParticlePacket, FancyParticlePacket>(FancyParticlePacket::class.java, FancyParticlePacket::class.java, 0, Side.CLIENT)
            instance.registerMessage<LightningEffectPacket, LightningEffectPacket>(LightningEffectPacket::class.java, LightningEffectPacket::class.java, 1, Side.CLIENT)
            instance.registerMessage<FancyLineParticlePacket, FancyLineParticlePacket>(FancyLineParticlePacket::class.java, FancyLineParticlePacket::class.java, 2, Side.CLIENT)
        } catch (server: NoClassDefFoundError) {
            server.printStackTrace()
        }

        //todo this is bad
        //Caused by: java.lang.NoClassDefFoundError: net/minecraft/client/entity/EntityPlayerSP
    }

    //lol
    fun tellEveryone(message: IMessage) {
        instance.sendToAll(message)
    }

    fun tellEveryoneAround(message: IMessage, dim: Int, x: Int, y: Int, z: Int, range: Int) {
        instance.sendToAllAround(message, NetworkRegistry.TargetPoint(dim, x.toDouble(), y.toDouble(), z.toDouble(), range.toDouble()))
    }

    fun tellEveryoneAround(message: IMessage, dim: Int, pos: BlockPos, range: Int) {
        instance.sendToAllAround(message, NetworkRegistry.TargetPoint(dim, pos.x.toDouble(), pos.y.toDouble(), pos.z.toDouble(), range.toDouble()))
    }

    fun tellEveryoneAround(message: IMessage, point: NetworkRegistry.TargetPoint) {
        instance.sendToAllAround(message, point)
    }
}
